### PR TITLE
Do not throw exception if byte array too short for fromBytes

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: api/all/src/main/java/io/opentelemetry/api/trace/SpanId.java [hunk 1]
```diff
@@ -62,18 +63,20 @@ public static boolean isValid(CharSequence spanId) {
...
    * Returns the lowercase hex (base16) representation of the {@code SpanId} converted from the
-   * given bytes representation, or {@link #getInvalid()} if input is {@code null}.
+   * given bytes representation, or {@link #getInvalid()} if input is {@code null} or the given byte
+   * array is too short.
+   *
+   * <p>It coverts the first 8 bytes of the array.
    *
    * @param spanIdBytes the bytes (8-byte array) representation of the {@code SpanId}.
    * @return the lowercase hex (base16) representation of the {@code SpanId}.
-   * @throws IndexOutOfBoundsException if {@code spanIdBytes} too short.
```
Comment (bogdandrutu) [inline api/all/src/main/java/io/opentelemetry/api/trace/SpanId.java:69]: We were actually throwing exception if the array was too long :)

File: api/all/src/main/java/io/opentelemetry/api/trace/SpanId.java [hunk 2]
```diff
@@ -62,18 +63,20 @@ public static boolean isValid(CharSequence spanId) {
 
   /**
    * Returns the lowercase hex (base16) representation of the {@code SpanId} converted from the
-   * given bytes representation, or {@link #getInvalid()} if input is {@code null}.
+   * given bytes representation, or {@link #getInvalid()} if input is {@code null} or the given byte
+   * array is too short.
+   *
+   * <p>It coverts the first 8 bytes of the array.
```
Comment (anuraaga) [inline api/all/src/main/java/io/opentelemetry/api/trace/SpanId.java:69]:
```suggestion
   * <p>It converts the first 8 bytes of the array.
```

**Review by anuraaga — APPROVED** [review]

File: api/all/src/main/java/io/opentelemetry/api/trace/SpanId.java (hunk 2 above)
Comment (bogdandrutu) [reply api/all/src/main/java/io/opentelemetry/api/trace/SpanId.java]: Done.

**Review by carlosalberto — APPROVED** [review]

File: api/all/src/test/java/io/opentelemetry/api/trace/TraceIdTest.java [hunk 3]
```diff
@@ -42,10 +42,14 @@ void fromLongs() {
...
-
     String traceId = "0102030405060708090a0b0c0d0e0f00";
     assertThat(TraceId.fromBytes(OtelEncodingUtils.bytesFromBase16(traceId, TraceId.getLength())))
         .isEqualTo(traceId);
   }
+
+  @Test
+  void fromBytes_Invalid() {
+    assertThat(SpanId.fromBytes(null)).isEqualTo(SpanId.getInvalid());
+    assertThat(SpanId.fromBytes(new byte[] {0, 1, 2, 3, 4})).isEqualTo(SpanId.getInvalid());
```
Comment (anuraaga) [inline api/all/src/test/java/io/opentelemetry/api/trace/TraceIdTest.java:53]:
```suggestion
    assertThat(TraceId.fromBytes(null)).isEqualTo(TraceId.getInvalid());
    assertThat(TraceId.fromBytes(new byte[] {0, 1, 2, 3, 4})).isEqualTo(TraceId.getInvalid());
```

**Review by anuraaga — COMMENTED** [review]
While @jkwatson often doesn't like the codecov report, the patch diff not being 100% prompted the closer look :P

File: api/all/src/test/java/io/opentelemetry/api/trace/TraceIdTest.java (hunk 3 above)
Comment (bogdandrutu) [reply api/all/src/test/java/io/opentelemetry/api/trace/TraceIdTest.java]: :)))

**Comment by bogdandrutu** [comment]
@anuraaga I tried :)))

**Pull request merged**